webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/WebSitesRequestRouter.java

```diff
@@ -90,1056 +95,1093 @@ public WebSiteSessionController createComponentSessionController(
-      } else if (function.startsWith("portlet")) {
-        FolderDetail webSitesCurrentFolder = scc.getFolder("0");
-        scc.setSessionTopic(webSitesCurrentFolder);
-        request.setAttribute("CurrentFolder", webSitesCurrentFolder);
-
-        if (flag.equals("Publisher") || flag.equals("Admin")) {
-          destination = "/webSites/jsp/listSitePortlet.jsp";
-        } else // reader
-        {
-          destination = "/webSites/jsp/listSite_readerPortlet.jsp";
-        }
-      } else if (function.startsWith("searchResult")) {
-        String id = request.getParameter("Id"); /* id de la publication */
-
-        String typeRequest = request.getParameter("Type");
-
-        if ("Publication".equals(typeRequest) || "Site".equals(typeRequest)) {
-          // recherche de l'url complete d'acces a la page
-          try {
-            SiteDetail sitedetail;
-            if ("Site".equals(typeRequest)) {
-              sitedetail = scc.getWebSite(id);
-            } else {
-              PublicationDetail pubDetail = scc.getPublicationDetail(id);
-              sitedetail = scc.getWebSite(pubDetail.getVersion());
-            }
-
-            destination = getWebSitesDestination(sitedetail, request, scc);
-          } catch (Exception e) {
-            SilverTrace.warn("webSites", "WebSitesRequestRouter.getDestination()",
-                "root.MSG_GEN_PARAM_VALUE", null, e);
-          }
-        } else if ("Node".equals(typeRequest)) {
-          destination = scc.getComponentUrl() + "listSite.jsp?Action=Search&Id=" + id;
-        }
-      } else if ("SuggestLink".equals(function)) {
-        String nomSite = request.getParameter("nomSite");
-        String description = request.getParameter("description");
-        String nomPage = request.getParameter("nomPage");
-        String auteur = request.getParameter("auteur");
-        String date = request.getParameter("date");
-        String listeIcones = request.getParameter("ListeIcones");
-
-        int begin = 0;
-        int end = listeIcones.indexOf(',', begin);
-        String listeMessage = "";
-
-        // parcours des icones
-        while (end != -1) {
-          String nom = listeIcones.substring(begin, end);
-          listeMessage += "- " + nom + "\n";
-          begin = end + 1;
-          end = listeIcones.indexOf(',', begin);
-        }
+              /* publications : classer le site dans les themes cochés */
+    ArrayList<String> arrayToClassify = new ArrayList<>();
+    boolean publish = false;
+    for (String idTopic : listOfTopics.split(",")) {
+      if (!idTopic.isEmpty()) {
+        scc.addPublicationToFolder(pubId, idTopic);
+        publish = true;
+      }
+    }
 
-        scc.notifyPublishers(auteur, nomSite, description, nomPage, listeMessage, date);
+    if (publish) {
+      arrayToClassify.add(site.getId());
+      scc.publish(arrayToClassify); // set etat du site a 1
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3ATrailingCommentCheck)